eclass/go-env.eclass

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: go-env.eclass
+# @MAINTAINER:
+# Flatcar Linux Maintainers <infra@flatcar-linux.org>
+# @AUTHOR:
+# Flatcar Linux Maintainers <infra@flatcar-linux.org>
+# @BLURB: Helper eclass for setting the Go compile environment. Required for cross-compiling.
+# @DESCRIPTION:
+# This eclass includes a helper function for setting the compile environment for Go ebuilds.
+# Intended to be called by other Go eclasses in an early build stage, e.g. src_unpack.
+
+if [[ -z ${_GO_ENV_ECLASS} ]]; then
+_GO_ENV_ECLASS=1
+
+inherit toolchain-funcs
+
+# @FUNCTION: go-env_set_compile_environment
+# @DESCRIPTION:
+# Set up basic compile environment: CC, CXX, and GOARCH.
+# Also carry over CFLAGS, LDFLAGS and friends.
+# Required for cross-compiling with crossdev.
+# If not set, host defaults will be used and the resulting binaries are host arch.
+# (e.g. "emerge-aarch64-cross-linux-gnu foo" run on x86_64 will emerge "foo" for x86_64
+#  instead of aarch64)
+go-env_set_compile_environment() {
+	local arch=$(tc-arch "${CHOST}}")
```

Comment:
I was about to submit it once more when I noticed this line too! Extra `}` here, but again, I think this can just be `$(tc-arch)` in the same manner.

Reply:
Good point; I checked `tc-ninja_magic_to_arch()` and it has the same defaults as `tc-endian`. Thanks for catching another spurious brace! I definitely need to read my PRs more carefully before submitting.
This worked in my test builds because `tc-ninja_magic_to_arch()` does pattern matching of `host`, so `aarch64*` matches against `aarch64-cross-linux-gnu}`. Great catch, thanks again!